splunklogger.js

```diff
@@ -169,16 +250,66 @@ SplunkLogger.prototype._initializeConfig = function(config) {
         ret.protocol = config.protocol || ret.protocol || defaultConfig.protocol;
         ret.level = config.level || ret.level || defaultConfig.level;
 
+        ret.maxRetries = config.maxRetries || ret.maxRetries || defaultConfig.maxRetries;
```

Comment:
You repeat this pattern a few times - it may be good to put them in a utility function.